src/ucs/memory/rcache.c

```diff
@@ -1154,50 +1172,11 @@ static void ucs_rcache_global_list_remove(ucs_rcache_t *rcache)
     ucs_async_pipe_destroy(&pipe);
 }
 
-static void ucs_rcache_vfs_show_inv_q_length(void *obj,
-                                             ucs_string_buffer_t *strb,
-                                             void *arg_ptr, uint64_t arg_u64)
+size_t ucs_rcache_distribution_get_num_bins()
 {
-    ucs_rcache_t *rcache = obj;
-    size_t rcache_inv_q_length;
+    size_t max_pow2 = ucs_roundup_pow2(ucs_global_opts.rcache_stat_max);
 
-    ucs_spin_lock(&rcache->lock);
-    rcache_inv_q_length = ucs_queue_length(&rcache->inv_q);
-    ucs_spin_unlock(&rcache->lock);
-
-    ucs_string_buffer_appendf(strb, "%zu\n", rcache_inv_q_length);
-}
-
-static void ucs_rcache_vfs_show_gc_list_length(void *obj,
-                                               ucs_string_buffer_t *strb,
-                                               void *arg_ptr, uint64_t arg_u64)
-{
-    ucs_rcache_t *rcache = obj;
-    unsigned long rcache_gc_list_length;
-
-    ucs_spin_lock(&rcache->lock);
-    rcache_gc_list_length = ucs_list_length(&rcache->gc_list);
-    ucs_spin_unlock(&rcache->lock);
-
-    ucs_string_buffer_appendf(strb, "%lu\n", rcache_gc_list_length);
-}
-
-static void ucs_rcache_vfs_init(ucs_rcache_t *rcache)
-{
-    ucs_vfs_obj_add_dir(NULL, rcache, "ucs/rcache/%s", rcache->name);
-    ucs_vfs_obj_add_ro_file(rcache, ucs_vfs_show_primitive,
-                            &rcache->num_regions, UCS_VFS_TYPE_ULONG,
-                            "num_regions");
-    ucs_vfs_obj_add_ro_file(rcache, ucs_vfs_show_primitive, &rcache->total_size,
-                            UCS_VFS_TYPE_SIZET, "total_size");
-    ucs_vfs_obj_add_ro_file(rcache, ucs_vfs_show_ulunits,
-                            &rcache->params.max_regions, 0, "max_regions");
-    ucs_vfs_obj_add_ro_file(rcache, ucs_vfs_show_memunits,
-                            &rcache->params.max_size, 0, "max_size");
-    ucs_vfs_obj_add_ro_file(rcache, ucs_rcache_vfs_show_inv_q_length, NULL, 0,
-                            "inv_q/length");
-    ucs_vfs_obj_add_ro_file(rcache, ucs_rcache_vfs_show_gc_list_length, NULL, 0,
-                            "gc_list/length");
+    return ucs_ilog2(max_pow2 / UCS_RCACHE_STAT_MIN_POW2) + 2;
```

Comment:
why +2?

Reply:
E.g. RCACHE_STAT_MIN=1024, RCACHE_STAT_MAX=2048, then we have three bins 0..1024, 1024..2048, 2048..inf